RocketChat/Rocket.Chat.ReactNative
# fix: Android status bar theme on the first open isn't working properly

## Proposed changes


I discovered that the status bar was not re-rendering correctly! So the solution was to add a useEffect to set the background and barStyle. The behavior is the same as expected when added to the props, and it fixes the bug on Android while still functioning correctly on iOS. The difference is that the useEffect re-renders one extra time after it is mounted.

## Issue(s)	

Closes https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/4889

## How to test or reproduce


This error it's happening on the first open at my Samsung Galaxy A31. It's working properly with Xiaomi Redmi 9 and with iOS devices.

>  - OS in light mode.
>  - Open RocketChat => status bar text is light (but should be dark)
>  - Change OS to dark mode via Quick Settings Tile
>  - RocketChat status bar text is now light ✅
>  - Change OS to light mode
>  - RocketChat status bar text is now dark ✅
>  - Close RocketChat and remove it from Recent Apps
>  - Open RocketChat again
>  - RocketChat status bar text is now light again (but should be dark)

## Screenshots

#### Before

https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/71a00fa5-920f-4a04-8c1a-acd945de657b


#### After

https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/d1edcf32-572d-403e-a7be-cad6dc6b1f38


## Types of changes




## Checklist



## Further comments

[TC-691]

[TC-691]: https://rocketchat.atlassian.net/browse/TC-691?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
